target/linux/ipq40xx/files-4.14/arch/arm/boot/dts/qcom-ipq4019-linksys_ea8300.dts

```diff
@@ -0,0 +1,394 @@
+						read-only;
+					};
+
+					partition@780000 {
+						label = "kernel";
+						reg = <0x780000 0x5800000>;
+					};
+
+					partition@a80000 {
+						label = "rootfs";
+						reg = <0xa80000 0x5500000>;
+					};
+
+					partition@5f80000 {
+						label = "alt_kernel";
+						reg = <0x5f80000 0x5800000>;
+					};
+
+					partition@6280000 {
+						label = "alt_rootfs";
+						reg = <0x6280000 0x5500000>;
+					};
+
+					partition@b780000 {
+						label = "sysdiag";
+						reg = <0xb780000 0x100000>;
+						read-only;
+					};
+
+					partition@b880000 {
+						label = "syscfg";
+						reg = <0xb880000 0x4680000>;
+						read-only;
+					};
+				};
+			};
+		};
+
+
+		mdio@90000 {			// Enables phys 0-4
+			status = "okay";
+		};
+
+		ess-switch@c000000 {
+			status = "okay";
+		};
+
+		ess-psgmii@98000 {
+			status = "okay";
+		};
+
+
+		wifi@a000000 {			// wifi0
+			status = "okay";
+			qcom,ath10k-calibration-variant = "linksys-ea8300-fcc";
+		};
+
+		wifi@a800000 {			// wifi1
+			status = "okay";
+			qcom,ath10k-calibration-variant = "linksys-ea8300-fcc";
+		};
+
+		pci@40000000 {			// pcie0
+			status = "okay";
+
+			perst-gpio = <&tlmm 38 GPIO_ACTIVE_LOW>;
+			wake-gpio = <&tlmm 50 GPIO_ACTIVE_LOW>;
+
+			bridge@0,0 {
+				reg = <0x00000000 0 0 0 0>;
+				#address-cells = <3>;
+				#size-cells = <2>;
+				ranges;
+
+				wifi2: wifi@1,0 {
+					compatible = "qcom,ath10k";
+					status = "okay";
+					reg = <0x00010000 0 0 0 0>;
+					qcom,ath10k-calibration-variant = "linksys-ea8300-fcc";
+				};
+			};
+		};
+
+
+		ssphy@9a000 {			// usb3_ss_phy
+			status = "okay";
+		};
+
+		hsphy@a6000 {			// usb3_hs_phy
+			status = "okay";
+		};
+
+		usb3@8af8800 {
+			status = "okay";
+		};
+
+		hsphy@a8000 {			// usb2_hs_phy
+			status = "okay";
+		};
+
+		usb2@60f8800 {
+			status = "okay";
+		};
+
+
+
+		tcsr@1949000 {
+			compatible = "qcom,tcsr";
+			reg = <0x1949000 0x100>;
+			qcom,wifi_glb_cfg = <TCSR_WIFI_GLB_CFG>;
+		};
+
+		tcsr@194b000 {
+			compatible = "qcom,tcsr";
+			reg = <0x194b000 0x100>;
+			qcom,usb-hsphy-mode-select = <TCSR_USB_HSPHY_HOST_MODE>;
+		};
+
+		ess_tcsr@1953000 {  // May not be needed for ipqess / qca8k
+			compatible = "qcom,tcsr";
+			reg = <0x1953000 0x1000>;
+			qcom,ess-interface-select = <TCSR_ESS_PSGMII>;
+		};
+
+		tcsr@1957000 {
+			compatible = "qcom,tcsr";
+			reg = <0x1957000 0x100>;
+			qcom,wifi_noc_memtype_m0_m2 = <TCSR_WIFI_NOC_MEMTYPE_M0_M2>;
+		};
+
+
+		dma@7884000 {			// blsp_dma
+			status = "okay";
+		};
+
+		dma@7984000 {			// qpic_bam
+			status = "okay";
+		};
+
+		dma@8e04000 {			// cryptobam
+			status = "okay";
+		};
+
+		edma@c080000 {
+			status = "okay";
+		};
+
+
+		pinctrl@1000000 {		// tlmm
+
+			serial_0_pins: serial0-pinmux {
+				pins = "gpio16", "gpio17";
+				function = "blsp_uart0";
+				bias-disable;
+			};
+
+			i2c_0_pins: i2c-0-pinmux {
+				pins = "gpio20", "gpio21";
+				function = "blsp_i2c0";
+				bias-disable;
+			};
+
+			nand_pins: nand-pins {
```

Comment:
bias are missing. You could copy the node from the qcom-ipq4019-fritzbox-7530.dts.

Reply:
Using the bias settings from that device here is troublesome. 

While "only a little green light" that fails as a result makes it a red flag for me. The EA8300, the reference board variant it is based on, `qcom-ipq40xx-ap.dk07.1.dtsi`, and those similar have a different set of NAND pins[1] than the fritzbox-7530 suggested. gpio61, listed in the Fritzbox as a NAND pin, is used for LED control on the EA8300.

In searching through the Linux sources for 4.14 and 4.19 both, as well as the OEM GPL drop for the EA8300, I have not been able to find evidence of bias for the NAND pins being explicitly set in the DTS. Given that U-Boot was able to successfully read from the NAND to boot and write to it to change the environment and boot counter, I would guess that it is setting up the bias properly. Linksys did not release the U-Boot source so I do not have access to this.

I'd rather not put in information that I can't confirm (bias for NAND pins) when so much gets copied "blindly" from device to device. 

As I don't have access to an IPQ4019 data sheet, is there information available that I could use to examine one or more registers to confirm what the bias settings are from the running device?

[1] See, for example
http://downloads.linksys.com/downloads/gpl/EA8300_v1.1.3.184925_GPL.tar.gz
https://elixir.bootlin.com/linux/v4.19.36/source/arch/arm/boot/dts/qcom-ipq4019-ap.dk07.1.dtsi
https://elixir.bootlin.com/linux/v4.19.36/source/arch/arm/boot/dts/qcom-ipq4019-ap.dk04.1.dtsi